apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Trace/TraceHeaderPropagatingHandlerTest.cs

```diff
@@ -0,0 +1,129 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Moq;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.Common.Tests
+{
+    public class TraceHeaderPropagatingHandlerTest
+    {
+        private const string traceId = "105445aa7843bc8bf206b12000100f00";
+        private const ulong spanId = 81237123;
+        private readonly TraceHeaderContext headerContext = 
+            TraceHeaderContext.Create(traceId, spanId, true);
+
+        private Mock<IManagedTracer> GetSetUpTracer()
+        {
+            var mockTracer = new Mock<IManagedTracer>();
+            mockTracer.Setup(t => t.GetCurrentTraceId()).Returns(traceId);
+            mockTracer.Setup(t => t.GetCurrentSpanId()).Returns(spanId);
+            return mockTracer;
+        }
+
+        [Fact]
+        public async Task SendAsync_NoTrace()
+        {
+            var mockTracer = new Mock<IManagedTracer>();
+            var traceHandler = TraceHeaderPropagatingHandler.Create(mockTracer.Object);
+            traceHandler.InnerHandler = new FakeDelegatingHandler();
+
+            using (var httpClient = new HttpClient(traceHandler))
+            {
+                await httpClient.GetAsync("https://www.google.com");
+            }
+
+            mockTracer.Verify(t => t.StartSpan(It.IsAny<string>(), null), Times.Never());
+            mockTracer.Verify(t => t.EndSpan(), Times.Never());
+        }
+
+        [Fact]
+        public async Task SendAsync_Trace()
+        {
+
+            var mockTracer = GetSetUpTracer();
+            var traceHandler = TraceHeaderPropagatingHandler.Create(mockTracer.Object);
+            traceHandler.InnerHandler = new FakeDelegatingHandler(headerContext);
+
+            var requestUri = new Uri("https://www.google.com");
+
+            using (var httpClient = new HttpClient(traceHandler))
+            {
+                await httpClient.GetAsync(requestUri);
+            }
+
+            mockTracer.Verify(t => t.StartSpan(requestUri.ToString(), null), Times.Once());
+            mockTracer.Verify(t => t.EndSpan(), Times.Once());
+        }
+
+        [Fact]
+        public async Task SendAsync_TraceException()
+        {
+            var mockTracer = GetSetUpTracer();
+            var traceHandler = TraceHeaderPropagatingHandler.Create(mockTracer.Object);
+            traceHandler.InnerHandler = new FakeDelegatingHandler(headerContext, true);
```

Comment:
This is a matter of taste, so feel free to ignore it: it took a little bit of can-you-spot-the-difference comparing between this test and the previous one to understand why this one is expecting an exception. I find boolean literal arguments to be very unhelpful for grokking code at a glance, so I usually like to use named arguments in these situations. I find
```csharp
... new FakeDelegatingHandler(headerContext, throwException: true);
```
to be much more self-documenting.